packages/framework-core/src/services/event-store.ts

```diff
@@ -17,90 +17,72 @@ const originOfTime = new Date(0).toISOString() // Unix epoch
 export class EventStore {
   public constructor(readonly config: BoosterConfig) {}
 
+  /**
+   * Will fetch the latest snapshot for an entity by applying a reduction
+   * since the time of creation of the last snapshot, UNLESS `pendingEnvelopes`
+   * are passed, in that case, it won't load the events, but rather it will
+   * proceed with the creation of the new snapshot by considering `pendingEnvelopes`
+   * as the latest events.
+   *
+   * Also, in order to make next calls faster, this method stores the calculated
+   * snapshot.
+   */
   public async fetchEntitySnapshot(entityName: string, entityID: UUID): Promise<EventEnvelope | null> {
```

Comment:
Should we rename this method to make it clear that we are going to create snapshots here?

Reply:
Ugh, let's better create an issue for this or we will never merge this change 😅